base/context.go

```diff
@@ -101,10 +114,15 @@ func (ctx *Context) GetClientTLSConfig() (*tls.Config, error) {
 	return ctx.clientTLSConfig, nil
 }
 
-// GetServerTLSConfig returns the context server TLS config, initializing it
-// if needed. It uses the context Certs field.
-// If Certs is empty, load insecure configs.
+// GetServerTLSConfig returns the context server TLS config, initializing it if needed.
+// If Insecure is true, return a nil config, otherwise load a config based
+// on the Certs directory. If Certs is empty, return a nil config.
```

Comment:
If Certs is empty it returns an error, not a nil config. (well, it returns both but the nil config doesn't really count)